version: '3.7'

volumes:
  grafana_data: {}
  mongo_data: {}
  prometheus_data: {}

networks:
  back-tier:
  front-tier:

services:
  prediction_service:
    image: gender-classifier:f37sltuursxkeqqb
    depends_on:
      - evidently_service
      - mongo
    environment:
      EVIDENTLY_SERVICE: "http://evidently_service.:8085"
      MONGODB_ADDRESS: "mongodb://mongo.:27017/"
    ports:
      - "3000:3000"
    networks:
      - back-tier
      - front-tier

  evidently_service:
    build:
      context: monitoring_service
      dockerfile: Dockerfile
    depends_on:
      - grafana
    volumes:
      - ./monitoring_service/datasets:/app/datasets
    ports:
      - "8085:8085"
    networks:
      - back-tier
      - front-tier

  prometheus:
    image: prom/prometheus
    volumes:
      - ./monitoring_service/config/prometheus.yml:/etc/prometheus/prometheus.yml
      - prometheus_data:/prometheus
    command:
      - '--config.file=/etc/prometheus/prometheus.yml'
      - '--storage.tsdb.path=/prometheus'
    ports:
      - "9090:9090"
    networks:
      - back-tier
    restart: always

  grafana:
    image: grafana/grafana
    user: "472"
    depends_on:
      - prometheus
    environment:
      #GF_LOG_LEVEL: "error"
      GF_SERVER_HTTP_PORT: 2345
    ports:
      - "2345:2345"
    volumes:
      - ./monitoring_service/config/grafana_datasources.yaml:/etc/grafana/provisioning/datasources/datasource.yaml:ro
      - ./monitoring_service/config/grafana_dashboards.yaml:/etc/grafana/provisioning/dashboards/dashboards.yaml:ro
      - ./monitoring_service/dashboards:/opt/grafana/dashboards
      - grafana_data:/var/lib/grafana
    networks:
      - back-tier
      - front-tier
    restart: always

  mongo:
    image: mongo
    volumes:
    - mongo_data:/data/db
    ports:
    - 27017:27017
    networks:
    - back-tier
    - front-tier